src/analysis/special_functions/gamma.lean

```diff
@@ -127,3 +133,204 @@ begin
 end
 
 end complex
+
+/- Now we establish the recurrence relation `Γ(s + 1) = s * Γ(s)` using integration by parts. -/
+
+namespace complex
+
+section Gamma_recurrence
+
+/-- The indefinite version of the Γ function, Γ(s, X) = ∫ x ∈ 0..X, exp(-x) x ^ (s - 1). -/
+def partial_Gamma (s : ℂ) (X : ℝ) : ℂ := ∫ x in 0..X, (-x).exp * x ^ (s - 1)
+
+lemma tendsto_partial_Gamma {s : ℂ} (hs: 1 ≤ s.re) :
+  tendsto (λ X:ℝ, partial_Gamma s X) at_top (𝓝 $ Gamma_integral s) :=
+interval_integral_tendsto_integral_Ioi 0 (Gamma_integral_convergent hs) tendsto_id
+
+lemma Gamma_integrand_interval_integrable (s : ℂ) {X : ℝ} (hs : 1 ≤ s.re) (hX : 0 ≤ X):
+  interval_integrable (λ x, (-x).exp * x ^ (s - 1) : ℝ → ℂ) volume 0 X :=
+begin
+  rw interval_integrable_iff_integrable_Ioc_of_le hX,
+  exact integrable_on.mono_set (Gamma_integral_convergent hs) Ioc_subset_Ioi_self
+end
+
+lemma Gamma_integrand_deriv_integrable_A {s : ℂ} (hs: 1 ≤ s.re) {X : ℝ} (hX : 0 ≤ X):
+ interval_integrable (λ x, -((-x).exp * x ^ s) : ℝ → ℂ) volume 0 X :=
+begin
+  have t := (Gamma_integrand_interval_integrable (s+1) _ hX).neg,
+  { simpa using t },
+  { simp only [add_re, one_re], linarith,},
+end
+
+lemma Gamma_integrand_deriv_integrable_B {s : ℂ} (hs: 1 ≤ s.re) {Y : ℝ} (hY : 0 ≤ Y) :
+  interval_integrable (λ (x : ℝ), (-x).exp * (s * x ^ (s - 1)) : ℝ → ℂ) volume 0 Y :=
+begin
+  have: (λ x, (-x).exp * (s * x ^ (s - 1)) : ℝ → ℂ) =
+    (λ x, s * ((-x).exp * x ^ (s - 1)) : ℝ → ℂ) := by { ext1, ring, },
+  rw [this, interval_integrable_iff_integrable_Ioc_of_le hY],
+  split,
+  { refine (continuous_on_const.mul _).ae_strongly_measurable measurable_set_Ioc,
+    apply (continuous_of_real.comp continuous_neg.exp).continuous_on.mul,
+    apply continuous_at.continuous_on,
+    intros x hx,
+    refine (_ : continuous_at (λ x:ℂ, x ^ (s - 1)) _).comp continuous_of_real.continuous_at,
+    apply continuous_at_cpow_const, rw of_real_re, exact or.inl hx.1, },
+  apply has_finite_integral_of_bounded, swap, exact s.abs * Y ^ (s.re - 1),
+  refine (ae_restrict_iff' measurable_set_Ioc).mpr (ae_of_all _ (λ x hx, _)),
+  rw [norm_eq_abs, abs_mul,abs_mul, abs_of_nonneg (exp_pos(-x)).le],
+  refine mul_le_mul_of_nonneg_left _ (abs_nonneg s),
+  have i1: (-x).exp ≤ 1 := by { simpa using hx.1.le, },
+  have i2: abs (↑x ^ (s - 1)) ≤ Y ^ (s.re - 1),
+  { rw [abs_cpow_eq_rpow_re_of_pos hx.1 _, sub_re, one_re],
+    apply rpow_le_rpow hx.1.le hx.2, linarith, },
+  simpa using mul_le_mul i1 i2 (abs_nonneg (↑x ^ (s - 1))) zero_le_one,
+end
+
+lemma partial_Gamma_recurrence {s : ℂ} (hs: 1 ≤ s.re) {X : ℝ} (hX : 0 ≤ X) :
```

Comment:
I understand this may seem pedantic, but on the scale of a large library following the naming scheme really pays off.
```suggestion
/-- The recurrence relation for the indefinite version of the Γ function. -/
lemma partial_Gamma_add_one {s : ℂ} (hs: 1 ≤ s.re) {X : ℝ} (hX : 0 ≤ X) :
```